verifier/token/cctp/attestation.go

```diff
@@ -2,42 +2,169 @@ package cctp
-		interval: config.AttestationAPIInterval,
-		cooldown: config.AttestationAPICooldown,
+) (AttestationService, error) {
+	client, err := NewHTTPClient(lggr, config)
+	if err != nil {
+		return nil, err
 	}
+	return &HTTPAttestationService{
+		lggr:               lggr,
+		client:             client,
+		ccvVerifierVersion: ccvVerifierVersion,
+		ccvAddresses:       config.ParsedVerifiers,
+	}, nil
 }
 
-func (h HTTPAttestationService) Fetch(
+// Fetch calls CCTP Attestation API using sourceChainDomain + txHash
+// It iterates through results to find the attestation for matching message
+// We use hookData and sender address to match the message against the protocol.Message.
+func (h *HTTPAttestationService) Fetch(
 	ctx context.Context,
 	txHash protocol.ByteSlice,
 	message protocol.Message,
-) (protocol.ByteSlice, error) {
-	// Call CCTPv2 using sourceChainDomain + txHash
-	// Find the attestation with the results set and return
-	// 		hookData can be found in attestation response - decodedMessage.decodedMessageBody.hookData
-	//      hookData format is the following <4 byte verifier version><32 byte msg ID>
-	return txHash, nil
+) (Attestation, error) {
+	sourceDomain, ok := sourceDomains[uint64(message.SourceChainSelector)]
+	if !ok {
+		return Attestation{}, fmt.Errorf("unsupported source chain selector: %d", message.SourceChainSelector)
+	}
+
+	response, err := h.client.GetMessages(ctx, message.SourceChainSelector, sourceDomain, txHash.String())
+	if err != nil {
+		return Attestation{}, fmt.Errorf(
+			"error fetching messages for chain selector %d and tx hash %s: %s",
+			message.SourceChainSelector, txHash, err,
+		)
+	}
+	return h.extractAttestationFromResponse(response, message)
+}
+
+func (h *HTTPAttestationService) extractAttestationFromResponse(response Messages, message protocol.Message) (Attestation, error) {
+	for _, msg := range response.Messages {
+		ok := h.matchesMessage(msg, message)
+		if !ok {
+			h.lggr.Debugw("skipping CCTP message as it doesn't match CCIP message", "message", msg)
+			continue
+		}
+		if msg.IsAttestationComplete() {
+			return NewAttestation(h.ccvVerifierVersion, msg)
+		}
+		return Attestation{}, fmt.Errorf("attestation is not ready")
+	}
+	return Attestation{}, fmt.Errorf("no matching message found in response")
+}
+
+func (h *HTTPAttestationService) matchesMessage(msg Message, message protocol.Message) bool {
+	messageID, err := message.MessageID()
+	if err != nil {
+		return false
+	}
+
+	lggr := logger.With(h.lggr, "cctpMessage", message, "messageID", messageID)
+
+	if !msg.IsV2() {
+		lggr.Debugw("CCTP Attestation: Skipping message due to unsupported CCTP version")
+		return false
+	}
+
+	ccvAddress, ok := h.ccvAddresses[message.SourceChainSelector]
+	if !ok {
+		lggr.Debugw("CCTP Attestation: No CCV address configured for source chain selector",
+			"sourceChainSelector", message.SourceChainSelector)
+		return false
+	}
+
+	senderAddress := protocol.UnknownAddress(msg.DecodedMessage.Sender)
```

Comment:
This is the sender we care about, i.e. the one that should equal ccvAddress for a given verifier version.
```suggestion
	senderAddress := protocol.UnknownAddress(msg.DecodedMessage.DecodedMessageBody.MessageSender)
```